Fluid/Filters/MiscFilters.cs

```diff
@@ -842,5 +847,77 @@ public static ValueTask<FluidValue> Sha256(FluidValue input, FilterArguments arg
                 return new StringValue(builder.ToString());
             }
         }
+
+        /// <summary>
+        /// https://shopify.dev/api/liquid/filters/additional-filters#default_pagination
+        /// </summary>
+        public static ValueTask<FluidValue> DefaultPagination(FluidValue input, FilterArguments arguments, TemplateContext context)
+        {
+            if (input.ToObjectValue() is not PaginateStatement.PaginateValue paginate) return StringValue.Empty;
+
+
+            using (var sb = StringBuilderPool.GetInstance())
+            {
+                var builder = sb.Builder;
+
+                if (paginate.Previous?.IsLink == true)
+                {
+                    var title = paginate.Previous.Title;
+
+                    if (arguments.HasNamed("previous"))
+                    {
+                        var str = arguments["previous"].ToStringValue();
+                        if (!string.IsNullOrWhiteSpace(str)) title = str;
+                    }
+
+                    builder.AppendFormat(
+                            "<span class=\"prev\"><a href=\"{0}\">{1}</a></span>",
+                            HttpUtility.HtmlAttributeEncode(paginate.Previous.Url),
+                            HttpUtility.HtmlEncode(title)
+                        );
+                }
+
+                if (paginate.Parts is not null and { Count: > 0 })
+                {
+                    foreach (var part in paginate.Parts)
+                    {
+                        if (part.IsLink)
+                        {
+                            builder.AppendFormat(
+                                "<span class=\"page\"><a href=\"{0}\">{1}</a></span>",
+                                HttpUtility.HtmlAttributeEncode(part.Url),
+                                HttpUtility.HtmlEncode(part.Title)
+                                );
+                        }
+                        else
+                        {
+                            builder.AppendFormat(
+                                "<span class=\"deco\">{0}</span>",
+                                HttpUtility.HtmlEncode(part.Title)
+                                );
+                        }
+                    }
+                }
+
+                if (paginate.Next?.IsLink == true)
+                {
+                    var title = paginate.Next.Title;
+
+                    if (arguments.HasNamed("next"))
+                    {
+                        var str = arguments["next"].ToStringValue();
+                        if (!string.IsNullOrWhiteSpace(str)) title = str;
+                    }
+
+                    builder.AppendFormat(
+                            "<span class=\"next\"><a href=\"{0}\">{1}</a></span>",
+                            HttpUtility.HtmlAttributeEncode(paginate.Previous.Url),
+                            HttpUtility.HtmlEncode(title)
```

Comment:
The encoder should also be part of the options if not already available somewhere else.